plugins/plugin-docker/che-plugin-docker-compose/src/main/java/org/eclipse/che/plugin/docker/compose/yaml/deserializer/CommandDeserializer.java

```diff
@@ -0,0 +1,111 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.compose.yaml.deserializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import org.eclipse.che.plugin.docker.compose.ComposeServiceImpl;
+import org.eclipse.che.plugin.docker.compose.yaml.ComposeEnvironmentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+
+/**
+ * Custom deserializer for parsing {@link ComposeServiceImpl#command}
+ * to list command words in the yaml compose file.
+ * <p>Command can be:</p>
+ * <ul>
+ * <li>string array:</li>
+ *
+ * <pre>
+ *  image: codenvy/mysql
+ *  environment:
+ *   MYSQL_USER: petclinic
+ *   MYSQL_PASSWORD: password
+ *  mem_limit: 2147483648
+ *  command: [service, mysql, start]
+ * </pre>
+ *
+ * <li>simple string line:</li>
+ *
+ * <pre>
+ *  image: codenvy/mysql
+ *  environment:
+ *   MYSQL_USER: petclinic
+ *   MYSQL_PASSWORD: password
+ *  mem_limit: 2147483648
+ *  command: service mysql start
+ * </pre>
+ * </ul>
+ *
+ *
+ * See more for parsing compose file
+ * {@link ComposeEnvironmentParser}.
+ * <p>Note: this deserializer works for json too.</p>
+ *
+ * @author Alexander Andrienko
+ */
+public class CommandDeserializer extends JsonDeserializer<List<String>> {
+
+    private static final String SPLIT_COMMAND_REGEX = "[ \n\r]+";
```

Comment:
Because user can have a long command and can break line. On Linux break line is "\n", earlier Mac "\r", Windows "\r\n"